modules/internal/ChapelBase.chpl

```diff
@@ -1099,12 +1102,21 @@ module ChapelBase {
 
   pragma "compiler generated"
   pragma "unalias fn"
-  inline proc chpl__unalias(ref x) { }
+  inline proc chpl__unalias(x) {
+    pragma "no copy" var ret = x;
+    return ret;
+  }
 
   pragma "unalias fn"
-  inline proc chpl__unalias(x:_iteratorClass) { }
+  inline proc chpl__unalias(x:_iteratorClass) {
+    pragma "no copy" var ret = x;
```

Comment:
could you add a comment here to indicate that this will return an array storing the result of the iterator?